101-bundle-json.md

```diff
@@ -500,4 +506,45 @@ The built-in actions (`install`, `upgrade`, `uninstall`) MUST NOT appear in the
 
 Implementations that do not support custom actions MUST NOT emit errors (either runtime or validation) if a bundle defines custom actions. That is, even if an implementation cannot execute custom actions, it MUST NOT fail to operate on bundles that declare custom actions.
 
+## Custom Extensions
+
+In many cases, the bundle descriptor is a sufficient artifact for delivering a CNAB bundle, since invocation images and other images may be retrieved from registries and repositories. Consequently, it is important to provide an extension mechanism. A _custom extension_ is a named collection of auxiliary data whose meaning is defined outside of this specification.
+
+Tools MAY define and declare additional fields inside of the `custom` section. Tools MUST NOT define additional fields anywhere else in the bundle descriptor. Implementations may produce an error or may ignore additional fields outside of the extension. However, implementations SHOULD preserve the data inside of the `custom` section even when that information is not understood by the implementation.
+
+The `custom` object is used as follows:
+
+```json
+{
+    "custom": {
+        "com.example.duffle-bag": {
+            "icon": "https://example.com/icon.png",
+            "iconType": "PNG"
+        },
+        "com.example.backup-preferences": {
+            "frequency": "daily"
+        }
+    }
+}
+```
+
+The format is:
+
+```json
+{
+    "extensions": {
+        "EXTENSION NAME": "ARBITRARY JSON DATA"
+    }
+}
+```
+
+The fields are defined as follows:
+
+- `extensions` defines the wrapper object for extensions
+  - EXTENSION NAME: a unique name for an extension. Names SHOULD follow the reverse DNS format `DOMAIN.ADDITIONAL.INFORMATION`, where DOMAIN conforms to the reverse DNS naming convention (`com.example` or `io.cnab`), and additional elements are separated by dots (`.`).
```

Comment:
Should we factorize this part so we can reference it wherever we need namespaced names? (custom actions, labels, extensions...)

Reply:
Done.